ci-operator/step-registry/baremetalds/devscripts/gather/baremetalds-devscripts-gather-commands.sh

```diff
@@ -55,20 +31,65 @@ echo "Get install-gather, if there is one..."
 cp /root/dev-scripts/ocp/*/log-bundle*.tar.gz /tmp/artifacts/log-bundle-\$HOSTNAME.tar.gz || true
 
 echo "Get sosreport including sar data..."
-sos report --case-id "\$HOSTNAME" --batch -o container_log,filesys,kvm,libvirt,logs,networkmanager,podman,processor,rpm,sar,virsh,dnf -k podman.all=on -k podman.logs=on --tmp-dir /tmp/artifacts
+sos report --case-id "\$HOSTNAME" --batch \
+  -o container_log,filesys,kvm,libvirt,logs,networkmanager,podman,processor,rpm,sar,virsh \
+  -k podman.all -k podman.logs \
+  --tmp-dir /tmp/artifacts
 
 echo "Get libvirt logs..."
 tar -czC "/var/log/libvirt/qemu" -f "/tmp/artifacts/libvirt-logs-\$HOSTNAME.tar.gz" --transform "s?^\.?libvirt-logs-\$HOSTNAME?" .
 
+. common.sh
+. network.sh
+. utils.sh
+
+# Pass master and workers IPs to installer-gather script to collect info from nodes which didn't join the cluster
+NODE_IPS=()
+for (( n=0; n<\$NUM_MASTERS; n++ ))
+do
+    node_name=\$(printf \$MASTER_HOSTNAME_FORMAT \$n)
+    node_ip=\$(sudo virsh net-dumpxml \$BAREMETAL_NETWORK_NAME | xmllint --xpath "string(//dns[*]/host/hostname[. = '\$node_name']/../@ip)" -)
+    NODE_IPS+=("\$node_ip")
+done
+for (( n=0; n<\$NUM_WORKERS; n++ ))
+do
+    node_name=\$(printf \$WORKER_HOSTNAME_FORMAT \$n)
+    node_ip=\$(sudo virsh net-dumpxml \$BAREMETAL_NETWORK_NAME | xmllint --xpath "string(//dns[*]/host/hostname[. = '\$node_name']/../@ip)" -)
+    NODE_IPS+=("\$node_ip")
+done
+for (( n=0; n<\$NUM_EXTRA_WORKERS; n++ ))
+do
+    node_name="extraworker-%d"
+    node_ip=\$(sudo virsh net-dumpxml \$BAREMETAL_NETWORK_NAME | xmllint --xpath "string(//dns[*]/host/hostname[. = '\$node_name']/../@ip)" -)
+    NODE_IPS+=("\$node_ip")
+done
+
+INTERNAL_SSH_OPTS=(-o 'ConnectTimeout=5'
+  -o 'StrictHostKeyChecking=no'
+  -o 'UserKnownHostsFile=/dev/null'
+  -o 'ServerAliveInterval=90'
+)
+
+# Collect sos report from each known node
+for NODE_IP in \${NODE_IPS[@]}; do
+  echo "Fetching SOS report from \${NODE_IP}"
+  ssh "\${INTERNAL_SSH_OPTS[@]}" core@\${NODE_IP} sudo mkdir /run/artifacts &&
+  ssh "\${INTERNAL_SSH_OPTS[@]}" core@\${NODE_IP} \
+    sudo podman run -it --name toolbox --authfile /var/lib/kubelet/config.json --privileged --ipc=host --net=host --pid=host -e HOST=/host -e NAME=toolbox- -e IMAGE=registry.redhat.io/rhel8/support-tools:latest -v /run:/run -v /var/log:/var/log -v /etc/machine-id:/etc/machine-id -v /etc/localtime:/etc/localtime -v /:/host registry.redhat.io/rhel8/support-tools:latest \
+        sos report --case-id "\$HOSTNAME" --batch \
+          -o container_log,filesys,logs,networkmanager,podman,processor,sar \
+          -k podman.all -k podman.logs \
+          --tmp-dir /run/artifacts && \
+  ssh "\${INTERNAL_SSH_OPTS[@]}" core@\${NODE_IP} sudo chown -R core:core /run/artifacts
+  scp "\${INTERNAL_SSH_OPTS[@]}" core@\${NODE_IP}:/run/artifacts/*.tar* /tmp/artifacts/ || true
```

Comment:
Not blocking, but by reviewing all the artifact names from a [sample job ](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshift_release/34664/rehearse-34664-pull-ci-openshift-installer-master-e2e-agent-compact-ipv4/1777346911766843392/artifacts/e2e-agent-compact-ipv4/baremetalds-devscripts-gather/artifacts/), I'm not sure that `$HOSTNAME` may really be useful, especially with many artifacts. Probably the job id / namespace could make more sense

For example:  
```
libvirt-logs-host-192-168-123-174.tar
sosreport-host-192-168-123-174-host-192-168-123-174-2024-04-08-kozgwgr.tar.xz
sosreport-host-192-168-123-174-host-192-168-123-174-2024-04-08-kozgwgr.tar.xz.sha256
sosreport-master-0-host-192-168-123-174-2024-04-08-gacjyoo.tar.xz
sosreport-master-0-host-192-168-123-174-2024-04-08-gacjyoo.tar.xz.sha256
sosreport-master-1-host-192-168-123-174-2024-04-08-ibkrzic.tar.xz
sosreport-master-1-host-192-168-123-174-2024-04-08-ibkrzic.tar.xz.sha256
sosreport-master-2-host-192-168-123-174-2024-04-08-ivqbfts.tar.xz
sosreport-master-2-host-192-168-123-174-2024-04-08-ivqbfts.tar.xz.sha256
```

Reply:
Hmm, I'd rather not duplicate job name/id in the filename - it may grow too large. Perhaps a hash of job name / id / hostname? It however makes it hard to link back to the original job ID anyway

Reply:
Maybe just removing the ip part could make more manageable? ie:
```
libvirt-logs-host.tar
sosreport-host-192-168-123-174-2024-04-08-kozgwgr.tar.xz
sosreport-host-192-168-123-174-2024-04-08-kozgwgr.tar.xz.sha256
sosreport-master-0-2024-04-08-gacjyoo.tar.xz
sosreport-master-0-2024-04-08-gacjyoo.tar.xz.sha256
sosreport-master-1-2024-04-08-ibkrzic.tar.xz
sosreport-master-1-2024-04-08-ibkrzic.tar.xz.sha256
sosreport-master-2-2024-04-08-ivqbfts.tar.xz
sosreport-master-2-04-08-ivqbfts.tar.xz.sha256
```
Anyhow, I don't want to block on that